### ISSUE TITLE
Windows 10 mkdir errors

### ISSUE THREAD

**Issue by SamFoster** [acquia/cli#485]
Hi folks,

I've downloaded 1.8.1 and am trying to use it on Win 10 (64 bit), i have php 8.0.3 running (and have tried with 7.3) and I keep getting a php mkdir error. It's exactly the same as this post 

https://stackoverflow.com/questions/65773529/how-to-install-acquai-cli-on-windows-machine

The phar file looks to get called OK - in my batch file calls acli and if I have no command it returns with a nice listing of possible commands - but if I try auth:login for example then I get the following....

--- snip---

```

PHP Warning:  mkdir(): No such file or directory in phar://C:/Users/Documents/acquia cloud ide/acli.phar/vendor/webmozart/json/src/JsonEncoder.php on line 234

Warning: mkdir(): No such file or directory in phar://C:/Users/Documents/acquia cloud ide/acli.phar/vendor/webmozart/json/src/JsonEncoder.php on line 234

In JsonFileStore.php line 325:

  Could not write \/.acquia/cloud_api.conf: Failed to open stream: No such file or directory (2)  


In JsonEncoder.php line 283:

  Could not write \/.acquia/cloud_api.conf: Failed to open stream: No such file or directory (2)  


auth:login [-k|--key KEY] [-s|--secret SECRET]


                                                                                                                        
 [help] You can find Acquia CLI documentation at https://docs.acquia.com/acquia-cli/                                    

```
--- snip -----

I've tried running in CMD, powershell and git bash - all the same error - even when I run them as an Administrator 

As I mentioned I do get a nice return of acli cmds I could call when I run 'acli' on its own.

Any ideas?

Thanks in advance

Sam

**Comment by grasmash** [comment]
It's trying to write a file to `~/.acquia/cloud_api.conf`. Apparently, it's not able to create the `~/.acquia` directory. This could be a permissions issue.

As a workaround, you could create the directory manually and ensure it's writable.

**Comment by SamFoster** [comment]
Matthew,

thanks for the initial thought. I'll do some digging as to where that would be on this Windows machine...

Sam

**Comment by SamFoster** [comment]
Folks

I found the problem - it was that the PC I am using did not have HOME environment variable set - at the moment I've set both SYSTEM and USER environment variables pointing to my home directory (which is in c:\users\ ) but I'll do some further testing to narrow it down. Suspect setting system environment variable HOME does it for all users, so probably don't need the user env variable.

**Comment by SamFoster** [comment]
I've managed to get an IDE created now, and am just familiarising myself with it. Great work folks!

I'm on Windows 10, 64 bit (ver 1.8.1 of the ACLI phar)  and I have seen further issues though that become apparent when I answer YES to questions such as 

Would you like Acquia CLI to search for a Cloud application that matches your local git config? (yes/no)

and 

Would you like to link the Cloud application xxxxxxx to this repository? (yes/no) [yes]:

I get errors such as 

```
  [Symfony\Component\Filesystem\Exception\IOException]
  Unable to write to the "phar://C:/Users/sam/Documents/acli/acli.phar" directory.
```

also if I answer N to the questions that give errors I eventually get a link to the acli and the drupal site - neither of which get successfully opened by the script - what I get is...
```

Opening your IDE in browser...
[notice] Command: where xdg-open [Exit: 1]
[notice] Command: where open [Exit: 1]
[notice] Command: where start [Exit: 1]
[warning] Could not find a browser on your local machine.
```

But if I then manually copy/paste the links I get the to the IDE

Hope this is useful info 

Sam

**Comment by danepowell** [comment]
We fixed the missing home dir on Windows long ago in https://github.com/acquia/cli/pull/187

You can see that it falls back to checking `HOMEPATH`: https://github.com/acquia/cli/blob/11ada6309eb49c1d2331f06a33407066500d7783/src/Helpers/LocalMachineHelper.php#L236

I thought HOMEPATH was [standard to every Windows environment](https://stackoverflow.com/questions/606483/what-is-the-meaning-of-these-windows-environment-variables-homedrive-homepath) (it's set at Explorer initialization.) How is it that you came to unset it?

The second error is https://github.com/acquia/cli/issues/395 and should be tracked in that issue. We are awaiting guidance on how to reproduce the issue (i.e. starting from a vanilla Windows VM).

**Comment by danepowell** [comment]
Closing due to lack of response.

**Comment by SamFoster** [comment]
Folks

Sorry for delay in responding back - I've been on leave etc etc.

I just wanted to feedback that I have found out why I was getting the issues reported here 

it so happens that I am in an organisation that uses system policies to set the homepath, home (not set) and other system environment variables such as homedrive. 

I've altered my batch file that calls acli.phar to just set them each time. I am aware there are probs more elegant ways, but this works fine for me.

So...my issues re the mkdir errors was due to me using a Windows laptop (could have been a desktop tho) which was part of a network and had the homepath etc enviroment variables set as a network drive - and as we are all working from home re COVID etc the laptop obviously wasn't able to write there!

I have spotted another issue concerning with rsync not worjing when trying to push/pull files etc but I'll log that in another ticket

Thanks for the support thus far

Sam
